### DISCUSSION TITLE
Use Auth0 with my nextron application

### DISCUSSION THREAD

**Discussion by alexis-piquet** [saltyshiomix/nextron#186] [Q&A]
Hi everyone, I would like to export a method named "getServerSideProps" for each pages I have.
I want do that because i would like to use the method exported by auth0 named "withPageAuthRequired".

I've prepared all of my components and pages as the following exemple:

``` typescript
import React from 'react'
import { NextPage } from 'next'
import DefaultLayout from '../../../components/layouts'
import UserList from '../../../components/model/user/list'
import { withPageAuthRequired } from '../../../share/auth0'

const UserSpace: NextPage = () => (
  <DefaultLayout>
    <UserList />
  </DefaultLayout>
)

export const getServerSideProps = withPageAuthRequired()

export default UserSpace

```
``` typescript
import { SERVICE_DATABASE } from '@front/share/constants'
import MappedTable from '@front/components/MappedTable'
import { useCollection } from '@front/share/hooks/collection'
import React from 'react'

const UserList: React.FC = () => {
  const propertyNames = [ '_id', 'name', 'description' ]
  const collection = useCollection({
    service: SERVICE_DATABASE,
    model: 'user'
  })
  return <MappedTable collection={collection} propertyNames={propertyNames} />
}

export default UserList
```
But when I do a "npm start" 
![image](https://user-images.githubusercontent.com/20417993/120748124-8c563b00-c502-11eb-885b-aa50b979c984.png)

I got this error (for all of my pages)
![image](https://user-images.githubusercontent.com/20417993/120748226-b6a7f880-c502-11eb-9fb1-162f99194b4e.png)

If I read [the documentation](https://nextjs.org/docs/messages/gssp-export), I can fix it with the usage of the script "next build" 
instead of "next build && next export".

So my question is, do you have any ideas how can I do that with a nextron project? Should I give to you more informations or code parts?

**Comment by developersaintt** [comment]
Hi @alexis-piquet , 

I came across the same issue, may i ask did you managed to find any solution?

Thanks
